src/py/flwr/cli/log.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower command line interface `log` command."""
+
+import time
+from logging import DEBUG, INFO
+
+import grpc
+import typer
+from typing_extensions import Annotated
+
+from flwr.common.grpc import GRPC_MAX_MESSAGE_LENGTH, create_channel
+from flwr.common.logger import log as logger
+
+
+def log(
+    run_id: Annotated[
+        int,
+        typer.Option(case_sensitive=False, help="The Flower run ID to query"),
+    ],
+    period: Annotated[
+        int,
+        typer.Option(
+            case_sensitive=False,
+            help="Use this to set connection refresh time period (in seconds)",
+        ),
+    ] = 60,
+    follow: Annotated[
+        bool,
+        typer.Option(case_sensitive=False, help="Use this flag to follow logstream"),
+    ] = True,
+) -> None:
+    """Get logs from Flower run."""
+
+    def on_channel_state_change(channel_connectivity: str) -> None:
+        """Log channel connectivity."""
+        logger(DEBUG, channel_connectivity)
+
+    # pylint: disable=unused-argument
+    def stream_logs(run_id: int, channel: grpc.Channel, duration: int) -> None:
+        """Stream logs from the beginning of a run with connection refresh."""
+
+    # pylint: disable=unused-argument
+    def print_logs(run_id: int, channel: grpc.Channel, timeout: int) -> None:
+        """Print logs from the beginning of a run."""
+
+    channel = create_channel(
+        server_address="127.0.0.1:9093",
```

Comment:
Maybe we could also leave this as an input arg? Also, I note that `:9093` is the default port for REST api (https://github.com/adap/flower/blob/c21c913bb248325dbbb91c47c1094a6bf9f8343b/src/py/flwr/server/app.py#L69). I wonder if we should use something else for the logs as default? @danielnugraha

Reply:
Yes, this can be an input arg. I'm conscious that the SuperExec address will also need to be set in `flwr run`. The address should be set in a consistent way later. In the meantime, I've added it to the input arg.